lib/jwt-utils.js

```diff
@@ -364,6 +364,14 @@ function checkPayload(token, cb) {
       return cb(errors.INVALID_EXP());
     }
 
+    var clientExpired = iatParsed + config.expiration;
+
+    if (clientExpired < payload.exp) {
```

Comment:
This condition is not really required. You can directly compare clientExpired with currentDate and the palyload.exp with currentDate, or select the more restrictive expiration and the compare with current date.